src/components/drawer/bl-drawer.stories.mdx

```diff
@@ -0,0 +1,111 @@
+  </p>
+`
+
+export const DrawerTemplate = (args) => html`
+  <bl-drawer id=${ifDefined(args.id)}
+             caption=${ifDefined(args.caption)}
+             embedUrl=${ifDefined(args.embedUrl)}
+             externalLink=${ifDefined(args.externalLink)}>
+  ${ifDefined(args.content)}
+  </bl-drawer>
+`
+
+export const StoryTemplate = (args) => html`
+  ${DrawerTemplate(args)}
+  <bl-button @click="${(event)=>openDialog(event,args)}">${ifDefined(args.buttonText)}</bl-button>
+`
+
+# Drawer
+
+A drawer presents context-specific information and/or actions without leaving the current page
+
+<bl-alert variant="warning" icon caption="Note">Inline styles in examples are only for **demo purposes**. Use regular CSS classes or tag selectors to set styles.</bl-alert>
+
+### Design Rules
+
+* Title and external link displayed on header section in drawer. Both of them are optional.
+* Close button always displayed on header section and drawer can be closed by clicking close button.
+* By default, Drawer can not close by clicking somewhere outside drawer.
+* Drawer appears right side on the page with full height expanded.
+* Title can be multiline automatically if it does not fit one line.
+* When drawer does not fit in its fixed size, it switches to mobile view.
+* There is an attribute about iframe and drawer component handle iframe rendering itself.
+* Only one drawer can display at the same time. When one drawer opens others will be closed.
+
+## Usage
+
+<Canvas>
+  <Story name="Default Values"
+         play={(event) => openDialog(event,{id:'drawer-1'})}
+         args={{id:"drawer-1", buttonText:"default drawer", content:DummyContent()}}>
+    {StoryTemplate.bind({})}
+  </Story>
+  <Story name="With caption"
+         play={(event)=> openDialog(event,{id:'drawer-2'})}
+         args={{id:"drawer-2", buttonText:"with caption", caption: "Caption", content:'example content'}}>
+    {StoryTemplate.bind({})}
+  </Story>
+  <Story name="With long caption"
+         play={(event) => openDialog(event,{id:'drawer-3'})}
+         args={{id:"drawer-3", buttonText:"with long caption", caption: "This drawer has a long caption and automatically handle it", content:'example content'}}>
+    {StoryTemplate.bind({})}
+  </Story>
+  <Story name="With caption and externalLink"
+         play={(event) => openDialog(event,{id:'drawer-4'})}
+         args={{id:"drawer-4", buttonText:"with caption and external", caption: "Caption", externalLink:"some-url", content:'example content'}}>
+    {StoryTemplate.bind({})}
+  </Story>
+  <Story name="With caption and embedUrl"
+         play={(event) => openDialog(event,{id:'drawer-5'})}
+         args={{id:"drawer-5", buttonText:"with caption, embedUrl", caption: "Caption", embedUrl:"some-url"}}>
```

Comment:
There might be issue here. I know chrome only allow iframe pages on same domain. I do not know how other browsers handle this case. So, what should we do?

Reply:
Browsers allow different domains. Iframe and CSP headers have some attributes to prevent this but by default, it's allowed. Maybe you can try to put a basic HTML file in drawer folder and try to open it with the iframe. We may need to do some extra configuration to include this HTML file in Storybook build though.

Reply:
Actually I tried some pages from different domains on playground(localhost). But I can not access. Then I have tried with another html file exists on localhost and it worked.